src/VisualStudio/Core/Def/Implementation/AbstractVsTextViewFilter`2.cs

```diff
@@ -47,18 +48,24 @@ int IVsTextViewFilter.GetDataTipText(TextSpan[] pSpan, out string pbstrText)
         }
 
         protected virtual int GetDataTipTextImpl(TextSpan[] pSpan, out string pbstrText)
+        {
+            var subjectBuffer = WpfTextView.GetBufferContainingCaret();
+            if (subjectBuffer == null)
+            {
+                pbstrText = null;
+                return VSConstants.E_FAIL;
+            }
+
+            return GetDataTipTextImpl(subjectBuffer, pSpan, out pbstrText);
+        }
+ 
+        protected int GetDataTipTextImpl(ITextBuffer subjectBuffer, TextSpan[] pSpan, out string pbstrText)
         {
             pbstrText = null;
 
             var debugInfo = LanguageService.LanguageDebugInfo;
             if (debugInfo != null)
```

Comment:
Seems like this check should be moved to  IVsTextViewFilter.GetDataTipText before we call GetDataTipTextImpl.

If there is no debug info there is no point in doing anything.

Reply:
Thank you! I do not want to fix this in servicing 15.7. I have created a work item [https://github.com/dotnet/roslyn/issues/28064](https://github.com/dotnet/roslyn/issues/28064) and will send a PR for Dev16.

---
In reply to: [197256969](https://github.com/dotnet/roslyn/pull/28060#discussion_r197256969) [](ancestors = 197256969)